Feature/#61 change vertical positioning

PhlppKrmr opened the pull request.

PhlppKrmr commented:
Fyi: Will add some testing in the coming days, just want you to check if it runs on your machines.

motschel123 reviewed (changes requested):
Works on my machine.

- Needs improvement on performance: 
   don't re-init lines of offset changes, but simply add up line offset In update function
- Try to touch as little code as possible to implement the new feature, **renaming, restructuring, refactoring should be done in separate** to make reviewing easier and prevent big merge conflicts in the future

PhlppKrmr commented:
@jenswaechtler , was a stupid mistake Marcel and me made, we removed the function I added to test something else. Should be working now.
Regarding your other point (Dragging): This was not scope of the issue, simply because we dont have the 0-indicator for now. AFAIK we want to introduce dragging by using this indicator.

nicolaskolbenschlag reviewed (approved):
Works!

leon-juenemann reviewed (approved):
Nice. Works for me.

PhlppKrmr commented:
Layout can me majorly improved, we should go for that next sprint.
